apps/src/templates/sectionProgressV2/SectionProgressV2.jsx

```diff
@@ -44,18 +46,25 @@ function SectionProgressV2({
     getLocalStorage(scriptId, sectionId)
   );
 
-  React.useEffect(
-    () => setExpandedLessonIds(getLocalStorage(scriptId, sectionId)),
-    [scriptId, sectionId]
-  );
+  React.useEffect(() => {
+    setExpandedLessonIds(getLocalStorage(scriptId, sectionId));
+    analyticsReporter.sendEvent(EVENTS.PROGRESS_V2_VIEW, {
+      sectionId: sectionId,
+      unitId: scriptId,
+    });
+  }, [scriptId, sectionId]);
 
   const setExpandedLessons = React.useCallback(
-    expandedLessonIds => {
-      setExpandedLessonIds(expandedLessonIds);
-      trySetLocalStorage(
-        getLocalStorageString(scriptId, sectionId),
-        JSON.stringify(expandedLessonIds)
-      );
+    updateFunction => {
```

Comment:
Modified from new value to an update function in order to prevent weird race condition bugs.